src/EditorFeatures/CSharpTest/CodeActions/ConvertIfToSwitch/ConvertIfToSwitchTests.cs

```diff
@@ -2650,5 +2648,68 @@ void M(int p)
                 CodeActionValidationMode = CodeActionValidationMode.None,
             }.RunAsync();
         }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsConvertIfToSwitch)]
+        public async Task TestMissingOnImplicitCastInRelationalPattern()
+        {
+            var source =
+@"class C
+{
+    void M(char c)
+    {
+        $$if (c >= 128 || c == 'a')
+            System.Console.WriteLine(c);
+    }
+}";
```

Comment:
I'd expect the refactoring to be offered, but it adds the previously implicit casts as explicit. The result I'm expecting is:

```csharp
class C
{
    void M(char c)
    {
        switch (c)
        {
            case >= (char)128:
            case 'a':
                System.Console.WriteLine(c);
                break;
        }   
    }
}
```

Reply:
See the comment in the analyzer. That changes the semantics, which I figured it was best to avoid